web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte

```diff
@@ -93,17 +181,20 @@
 
 {#if assets.length > 0}
 	<div class="flex flex-wrap gap-1 w-full pb-20" bind:clientWidth={viewWidth}>
-		{#each assets as asset (asset.id)}
-			<Thumbnail
-				{asset}
-				{thumbnailSize}
-				readonly={disableAssetSelect}
-				publicSharedKey={sharedLink?.key}
-				format={assets.length < 7 ? ThumbnailFormat.Jpeg : ThumbnailFormat.Webp}
-				on:click={(e) => (isMultiSelectionMode ? selectAssetHandler(e) : viewAssetHandler(e))}
-				on:select={selectAssetHandler}
-				selected={selectedAssets.has(asset)}
-			/>
+		{#each assets as asset, index (asset.id)}
+			{#if justifiedLayoutResult?.boxes != null && justifiedLayoutResult?.boxes.length > 0 && justifiedLayoutResult.boxes[index] != null}
+				<Thumbnail
+					{asset}
+					thumbnailWidth={justifiedLayoutResult.boxes[index].width || 235}
+					thumbnailHeight={justifiedLayoutResult.boxes[index].height || 235}
+					readonly={disableAssetSelect}
+					publicSharedKey={sharedLink?.key}
+					format={assets.length < 7 ? ThumbnailFormat.Jpeg : ThumbnailFormat.Webp}
+					on:click={(e) => (isMultiSelectionMode ? selectAssetHandler(e) : viewAssetHandler(e))}
+					on:select={selectAssetHandler}
+					selected={selectedAssets.has(asset)}
+				/>
+			{/if}
 		{/each}
 	</div>
```

Comment:
Would probably be cleaner to create the justified layout like this. Then you don't need as much boilerplate and reactivity is handled for you. An example:

```svelte
<div ... bind:clientWidth={viewWidth}>
	{#if viewWidth}
		{@const geoArray = assets.map(getAssetRatio)}
		{@const justifiedLayout = createJustifiedLayout(geoArray, {
			targetRowHeight: 235,
			containerWidth: viewWidth
		})}
		{#each assets as asset, index (asset.id)}
			<Thumbnail width={justifiedLayout.boxes[index].width || 235} ... />
		{/each}
	{/if}
</div>
```

Or declare variables in script if you prefer:
```svelte
$: geoArray = ...
$: justifiedLayout = ...
```